p2p/switch.go

```diff
@@ -281,8 +281,11 @@ func (sw *Switch) StopPeerForError(peer Peer, reason interface{}) {
 	sw.stopAndRemovePeer(peer, reason)
 
 	if peer.IsPersistent() {
-		// NOTE: this is the self-reported addr, not the original we dialed
-		go sw.reconnectToPeer(peer.NodeInfo().NetAddress())
+		addr := peer.OriginalAddr()
+		if addr == nil {
+			panic(fmt.Sprintf("persistent peer %v with no original address", peer))
```

Comment:
is it documented somewhere that only outbound peers can be persistent on our end?

Reply:
No. Actually, should we try and dial self-reported addr if persistent peer is "inbound"?

Reply:
https://github.com/tendermint/tendermint/pull/2031